src/vehicle.cpp

```diff
@@ -3815,8 +3857,88 @@ void vehicle::idle(bool on_map) {
 
     if( on_map ) {
         update_time();
-        if(scoop_on){
-            operate_scoop();
```

Comment:
Moving `operate_scoop` into `on_move` makes the scoop only work when the vehicles moves by itself. Previously it would work when the vehicle was standing still or when it was pulled/pushed by the character. That behavior was quite useful and realistic.